orchestrator/db/filters/process.py

```diff
@@ -0,0 +1,130 @@
+from orchestrator.db import ProcessSubscriptionTable, ProcessTable, ProductTable, SubscriptionTable, db
+from orchestrator.db.database import SearchQuery
+from orchestrator.db.filters.filters import generic_filter
+
+logger = structlog.get_logger(__name__)
+
+
+def pid_filter(query: SearchQuery, value: str) -> SearchQuery:
+    return query.filter(cast(ProcessTable.pid, String).ilike("%" + value + "%"))
+
+
+def is_task_filter(query: SearchQuery, value: str) -> SearchQuery:
+    value_as_bool = value.lower() in ("yes", "y", "ye", "true", "1", "ja")
+    return query.filter(ProcessTable.is_task.is_(value_as_bool))
+
+
+def assignee_filter(query: SearchQuery, value: str) -> SearchQuery:
+    assignees = value.split("-")
+    return query.filter(ProcessTable.assignee.in_(assignees))
+
+
+def status_filter(query: SearchQuery, value: str) -> SearchQuery:
+    statuses = value.split("-")
+    return query.filter(ProcessTable.last_status.in_(statuses))
+
+
+def workflow_filter(query: SearchQuery, value: str) -> SearchQuery:
+    return query.filter(ProcessTable.workflow.ilike("%" + value + "%"))
+
+
+def creator_filter(query: SearchQuery, value: str) -> SearchQuery:
+    return query.filter(ProcessTable.created_by.ilike("%" + value + "%"))
+
+
+def organisation_filter(query: SearchQuery, value: str) -> SearchQuery:
+    try:
+        value_as_uuid = UUID(value)
+    except (ValueError, AttributeError):
+        msg = f"Not a valid organisation, must be a UUID: '{value}'"
+        logger.debug(msg)
+        raise_status(HTTPStatus.BAD_REQUEST, msg)
+
+    process_subscriptions = (
+        db.session.query(ProcessSubscriptionTable)
+        .join(SubscriptionTable)
+        .filter(SubscriptionTable.customer_id == value_as_uuid)
+        .subquery()
+    )
+    return query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+
+
+def product_filter(query: SearchQuery, value: str) -> SearchQuery:
+    process_subscriptions = (
+        db.session.query(ProcessSubscriptionTable)
+        .join(SubscriptionTable, ProductTable)
+        .filter(ProductTable.name.ilike("%" + value + "%"))
+        .subquery()
+    )
+    return query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+
+
+def tag_filter(query: SearchQuery, value: str) -> SearchQuery:
+    tags = value.split("-")
+    process_subscriptions = (
+        db.session.query(ProcessSubscriptionTable)
+        .join(SubscriptionTable, ProductTable)
+        .filter(ProductTable.tag.in_(tags))
+        .subquery()
+    )
+    return query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+
+
+def subscriptions_filter(query: SearchQuery, value: str) -> SearchQuery:
+    process_subscriptions = (
+        db.session.query(ProcessSubscriptionTable)
+        .join(SubscriptionTable)
+        .filter(SubscriptionTable.description.ilike("%" + value + "%"))
+        .subquery()
+    )
+    return query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+
```

Comment:
I still think these subqueries shouldn't be necessary since the base query already joins all tables that we need. But we can improve that later on.